qp_klp/Amplicon.py

```diff
@@ -81,23 +83,43 @@ def quality_control(self):
                 new_path = join(output_folder, basename(raw_fastq_file))
                 copyfile(raw_fastq_file, new_path)
 
-    def generate_reports(self, input_file_path):
-        super()._generate_reports(self.pipeline.mapping_file)
+    def generate_reports(self):
+        super()._generate_reports()
         return None  # amplicon doesn't need project names
 
+    def generate_touched_studies(self, qclient):
+        for study_id in self.prep_file_paths:
+            for prep_file_path in self.prep_file_paths[study_id]:
+                metadata = Step.parse_prep_file(prep_file_path)
+                if 'target_gene' in metadata[list(metadata.keys())[0]]:
+                    tg = metadata[list(metadata.keys())[0]]['target_gene']
+                    data_type = None
+                    for key in Step.AMPLICON_SUB_TYPES:
+                        if key in tg:
+                            data_type = key
+
+                    if data_type is None:
```

Comment:
This code seems to assume that `['target_gene']` can have multiple values in a single prep, right? However, it should only be one; if there are multiple that should be an error.

Reply:
We do assume that there will only be one type, but what I've noticed is that the value is often not '16S', '18S', or 'ITS' but a value where one of those will be a substring within. Hence the need to loop through the enumerated types and check.